clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -3157,7 +3177,15 @@ RValue CodeGenFunction::EmitBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
     if (Result->getType() != ResultType)
```

Comment:
Not sure if the way this is handling isCLZForZeroUndef() is what you want?

Reply:
Since in the additions below I check if the argument itself is zero instead of checking the result of `ctlz`, I don't see why the `ctlz` result for zero being undefined would be a problem.

Reply:
I was more thinking of the opposite: we don't need to make the result of the clz defined if we're not using it.

Reply:
Right. Is it worth making tests for this with a target where `isCLZForZeroUndef()` is false?